text/0042-partial-fill-rules.md

```diff
@@ -0,0 +1,296 @@
+1. If `partial_fill_change` is present, then decrypting the amount and token id using `amount_shared_secret` must succeed. The partial fill change
+   may not have a 0 value (to help clients avoid division by zero).
+1. If `partial_fill_change` is present, then a corresponding TxOut called the `fractional_change_output` must exist in the `TxPrefix.outputs`.
+   This must match the `partial_fill_change` in every field except possibly the `masked_amount`.
+1. The `fractional_change_output` must decrypt successfully using the `amount_shared_secret` of the `partial_fill_change`, and
+   it's amount must have the same token id as the `partial_fill_change`, and its value must be less or equal.
+1. The fill fraction is inferred at this point.
+   * The numerator is `partial_fill_change.value - fractional_change_output.value`.
+   * The denominator is `partial_fill_change.value`.
+1. The minimum fill rule is enforced: `partial_fill_change.value - fractional_change_output.value >= min_partial_fill_value`.
+   If this inequality does not hold, then the transaction is rejected for reasons of not meeting the minimum partial fill prescribed
+   by the origiantor.
+1. For each `partial_fill_output`:
+   * It must decrypt succesfully using the recorded `amount_shared_secret`.
+   * There must be a corresponding `fractional_output` among the `TxPrefix.outputs` which matches it in every field except possibly
+     the `masked_amount`.
+   * The fractional output must decrypt successfully using the `partial_fill_output`'s `amount_shared_secret`, and must match the
+     token id of the `partial_fill_output`.
+   * It must be the case that `fractional_output.value >= n/d * partial_fill_output.value` where `n` and `d` are the fill fraction
+     numerator and denominator. This equation must be validated by clearing denominators and checking an inequality of `u128`'s, to
+     avoid numerical issues.
+     
+     ```
+     fractional_output.value * fill_fraction_denominator >= fill_fraction_numerator * partial_fill_output.value
+     ```
+     
+     If this inequality does not hold, then the transaction must be rejected for reasons of not respecting the fill fraction.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This proposal has a few drawbacks.
+
+* The amounts of partial fill transactions are revealed to the enclave. Currently, MobileCoin transactions are validated using RingCT and do not reveal amounts to the enclave. It would be better if there were a way to validate partial fills without revealing this to the enclave, but for the use-cases envisioned, this data already has to be revealed anyways, so the use-cases aren't negatively impacted. We could imagine that an exchange service is based on SGX, and so the amounts of these quotes are only revealed inside of SGX, and are matched inside of SGX. Even in this case, it is still the case that if you penetrate SGX you will see the amounts of the SCI quotes.
```

Comment:
:upside_down_face: : 
```suggestion
* The amounts of partial fill transactions are revealed to the enclave. Currently, MobileCoin transactions are validated using RingCT and do not reveal amounts to the enclave. It would be better if there were a way to validate partial fills without revealing this to the enclave, but for the use cases envisioned, this data already has to be revealed anyways, so the use cases aren't negatively impacted. We could imagine that an exchange service is based on SGX, and so the amounts of these quotes are only revealed inside of SGX, and are matched inside of SGX. Even in this case, it is still the case that if you penetrate SGX you will see the amounts of the SCI quotes.
```